### PR TITLE
Add mock oracle contract

### PR DESCRIPTION
**Pull request by PatrickAlphaC**
So that we can test API calls effectively. It's a little different from the main oracle contract, in that it allows the contract to call

`mockOracle.fulfillOracleRequest(requestId, get_data, {from: User)`

When testing, instead of having to hash the result as done in the main oracle contract.

addresses #4095

### CODE REVIEW

**Review by se3000 — DISMISSED** [review]

File: evm-contracts/src/v0.6/tests/MockOracle.sol [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  event CancelOracleRequest(
+    bytes32 indexed requestId
+  );
+
+  /**
+   * @notice Deploy with the address of the LINK token
+   * @dev Sets the LinkToken address for the imported LinkTokenInterface
+   * @param _link The address of the LINK token
+   */
+  constructor(address _link)
```
Comment (alexroan) [inline evm-contracts/src/v0.6/tests/MockOracle.sol:48]: Separate line for parameter
```suggestion
  constructor(
    address _link
  )
```

File: evm-contracts/src/v0.6/tests/MockOracle.sol [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+   * @param _data The data to return to the consuming contract
+   * @return Status if the external call was successful
+   */
+  function fulfillOracleRequest(
+    bytes32 _requestId,
+    bytes32 _data
+  )
+    external
+    isValidRequest(_requestId)
+    returns (bool)
```
Comment (alexroan) [inline evm-contracts/src/v0.6/tests/MockOracle.sol:119]: Separate line per return value.
```suggestion
    returns (
      bool
    )
```

File: evm-contracts/src/v0.6/tests/MockOracle.sol [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+  /**
+   * @notice Returns the address of the LINK token
+   * @dev This is the public implementation for chainlinkTokenAddress, which is
+   * an internal method of the ChainlinkClient contract
+   */
+  function getChainlinkToken()
+    public
+    view
+    override
+    returns (address)
```
Comment (alexroan) [inline evm-contracts/src/v0.6/tests/MockOracle.sol:168]: Separate line per return value.
```suggestion
    returns (
      address
    )
```

File: evm-contracts/src/v0.6/tests/MockOracle.sol [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    return address(LinkToken);
+  }
+
+  // MODIFIERS
+
+  /**
+   * @dev Reverts if request ID does not exist
+   * @param _requestId The given request ID to check in stored `commitments`
+   */
+  modifier isValidRequest(bytes32 _requestId) {
```
Comment (alexroan) [inline evm-contracts/src/v0.6/tests/MockOracle.sol:179]: Separate line per parameter.
```suggestion
  modifier isValidRequest(
    bytes32 _requestId
  ) {
```

File: evm-contracts/src/v0.6/tests/MockOracle.sol [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+    require(commitments[_requestId].callbackAddr != address(0), "Must have a valid requestId");
+    _;
+  }
+
+
+  /**
+   * @dev Reverts if the callback address is the LINK token
+   * @param _to The callback address
+   */
+  modifier checkCallbackAddress(address _to) {
```
Comment (alexroan) [inline evm-contracts/src/v0.6/tests/MockOracle.sol:189]: Separate line per parameter.
```suggestion
  modifier checkCallbackAddress(
    address _to
  ) {
```

**Review by alexroan — CHANGES_REQUESTED** [review]
Just a few styling fixes.

**Review by alexroan — APPROVED** [review]

**Comment by PatrickAlphaC** [comment]
yyyyyyeessssss

**Pull request merged**